.gitignore

```diff
@@ -103,6 +103,8 @@ deps/npm/node_modules/.bin/
 /SHASUMS*.txt*
 
 # test artifacts
+tools/doc/node_modules
```

Comment:
So I think this should be 

```
tools/doc/node_modules/*
!tools/doc/node_modules/.bin
!tools/doc/node_modules/js-yaml/index.js
!tools/doc/node_modules/marked
```

And commit the package-lock.json into the source instead of ignoring it.